docs/02-app/02-api-reference/02-file-conventions/middleware.mdx

```diff
@@ -0,0 +1,63 @@
+---
+title: middleware.js
+description: API reference for the middleware.js  file.
+related:
+  title: Learn more about Middleware
+  links:
+    - app/building-your-application/routing/middleware
+---
+
+The middleware.ts file in Next.js is used to run code on the server before a request is finished. It provides the ability to modify the response based on the incoming request. This can include operations like URL rewriting, redirecting, altering request or response headers, or even sending a direct response.
+
+Middleware executes before the content is served from the cache and before routes are matched. It's particularly useful for implementing custom server-side functionalities like authentication, logging, or handling specific path patterns.
+
+Use the file `middleware.ts` (or .js) in the root of your project to define Middleware. For example, at the same level as pages or app, or inside src if applicable.
+
+```tsx filename="middleware.ts" switcher
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+
+// This function can be marked `async` if using `await` inside
+export function middleware(request: NextRequest) {
+  return NextResponse.redirect(new URL('/home', request.url))
+}
+
+export const config = {
+  matcher: '/about/:path*',
+}
+```
+
+```jsx filename="middleware.js" switcher
+import { NextResponse } from 'next/server'
+
+// This function can be marked `async` if using `await` inside
+export function middleware(request) {
+  return NextResponse.redirect(new URL('/home', request.url))
+}
+
+export const config = {
+  matcher: '/about/:path*',
+}
+```
+
+## Matcher
```

Comment:
Perhaps we can create a section called **Exports**, and talk about how:

 1. The middleware file **must** export a default function or a named `middleware` function, and the params it receives. 
 2. You can export an **optional** config object (which includes the matcher key, but also other options). We should include all the other options here in the reference.

Reply:
Thanks for the feedback! Made these changes, let me know what you think :)